### PR TITLE
Adds additional tabs for timdex sources

### PR DESCRIPTION
**Pull request by JPrevost**
Why are these changes being introduced:

* All sources should exist in at least one tab

Relevant ticket(s):

* https://mitlibraries.atlassian.net/browse/USE-517

How does this address that need:

* Adds additional tabs for timdex sources
* Adds additional blurbs for the new tabs

##### Accessibility

- [ ] ANDI or WAVE has been run in accordance to [our guide](https://mitlibraries.github.io/guides/basics/a11y.html).
- [ ] This PR contains no changes to the view layer.
- [ ] New issues flagged by ANDI or WAVE have been resolved.
- [ ] New issues flagged by ANDI or WAVE have been ticketed (link in the Pull Request details above).
- [ ] No new accessibility issues have been flagged.

##### New ENV

- [ ] All new ENV is documented in README.
- [ ] All new ENV has been added to Heroku Pipeline, Staging and Prod.
- [ ] ENV has not changed.

##### Approval beyond code review

- [ ] UXWS/stakeholder approval has been confirmed.
- [ ] UXWS/stakeholder review will be completed retroactively.
- [ ] UXWS/stakeholder review is not needed.

##### Additional context needed to review

E.g., if the PR includes updated dependencies and/or data
migration, or how to confirm the feature is working.

##### Code

- [ ] I have confirmed that the code works as intended.
- [ ] Any CodeClimate issues have been fixed or confirmed as
added technical debt.

##### Documentation

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message).
- [ ] The documentation has been updated or is unnecessary.
- [ ] New dependencies are appropriate or there were no changes.

##### Testing

- [ ] There are appropriate tests covering any new functionality.
- [ ] No additional test coverage is required.

### CODE REVIEW

File: test/helpers/results_helper_test.rb [hunk 1]
```diff
@@ -21,37 +21,10 @@ class ResultsHelperTest < ActionView::TestCase
...
-    assert_equal description, tab_description
-
-    params[:tab] = 'timdex'
-    description = 'Digital collections, images, documents, and more from MIT Libraries'
-    assert_equal description, tab_description
-
-    params[:tab] = 'website'
-    description = 'Information about the library: events, news, services, and more'
-    assert_equal description, tab_description
+    ResultsHelper::TAB_DESCRIPTIONS.each do |tab, expected|
```
Comment (matt-bernhardt) [inline test/helpers/results_helper_test.rb:24]: I like the efficiency of this test, but I wonder whether we should have a test for the behavior when a value outside that `TAB_DESCRIPTIONS` construct is presented?

More theoretically, I get that this test makes things more robust because we don't need to manually keep the list of test values in sync with the content of the helper method. This feels like a win - but I'm not sure where the line is where we just end up testing that `.each()` does what we expect.

File: app/helpers/results_helper.rb [hunk 2]
```diff
@@ -1,27 +1,34 @@
 module ResultsHelper
+  # Descriptions for each tab. HTML entries use raw anchor tags (all URLs are
+  # hardcoded external links, so no XSS risk) and must be marked html_safe.
+  TAB_DESCRIPTIONS = {
```
Comment (matt-bernhardt) [inline app/helpers/results_helper.rb:4]: I like this change in pattern a lot. 🍻

File: app/helpers/results_helper.rb [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
-    when 'aspace'
-      'Archives, manuscripts, and other unique materials related to MIT'
-    when 'timdex'
-      'Digital collections, images, documents, and more from MIT Libraries'
-    when 'website'
-      'Information about the library: events, news, services, and more'
-    else
+    TAB_DESCRIPTIONS.fetch(params[:tab]) do
       Rails.logger.error "Unknown tab parameter in `tab_description` helper: #{params[:tab]}"
+      ''
```
Comment (matt-bernhardt) [inline app/helpers/results_helper.rb:36]: Making sure I understand - this is the default value in case `tab_description` gets presented a value that we don't expect, right? So if there's a regression or mismatch somehow, we're not presenting gibberish, just an empty string.

**Review by matt-bernhardt — APPROVED** [review]
I have a non-blocking request for a different test type, but it isn't something I'm going to frame as a full request for changes - if there's something in this that I'm missing, I'm happy for this to merge as-is (the type of test I'm talking about didn't exist before, so this isn't a regression)

**Comment by djanelle-mit** [comment]
From my perspective, this looks good to go. Having a const to control the max viewable tabs makes sense. Since the truncation of tabs happens on pretty wide screens now, I should probably adjust the Tablet+ breakpoints to have a more traditional dropdown menu for the "more" overflow instead of the full width menu.

We can create a follow-up ticket for that though, and definitely shouldn't block things.

File: app/helpers/results_helper.rb (hunk 3 above)
Comment (JPrevost) [reply app/helpers/results_helper.rb:36]: Correct. Previous to this we were inadvertently returning `true` as the logging statement is apparently truthy.

File: test/helpers/results_helper_test.rb (hunk 1 above)
Comment (JPrevost) [reply test/helpers/results_helper_test.rb:24]: I definitely agree we should confirm something outside of the expected list returns blank. I'll confirm that test exists or add one.

Comment (JPrevost) [reply test/helpers/results_helper_test.rb:24]: I've added additional tests as suggested here. Thanks for pointing out we were missing edge cases!

**Review by matt-bernhardt — APPROVED** [review]
Yep, I think this looks good. Thanks!
